tests/python_package_test/test_engine.py

```diff
@@ -453,3 +453,17 @@ def test_pandas_categorical(self):
         np.testing.assert_almost_equal(pred0, pred2)
         np.testing.assert_almost_equal(pred0, pred3)
         np.testing.assert_almost_equal(pred0, pred4)
+
+    def test_subset_train_val(self):
+        '''
+        Tests that it's fine to construct a single lgb.Dataframe object,
+        takes subsets of it, and uses the subsets for training and validation
+        '''
+        n = 1000
+        X = np.random.normal(size=n * 2).reshape(n, 2)
+        y = np.random.normal(size=n)
+        tmp_dat = lgb.Dataset(X, y)
+        tmp_dat_train = tmp_dat.subset(np.arange(int(n * .8)))
+        tmp_dat_val = tmp_dat.subset(np.arange(int(n * .8), n))
```

Comment:
I think so.

I interpreted your last comment as suggesting I should replace the current test in this PR with one that explicitly tests the ref_chain logic instead of testing whether you can train a model with train / val as subsets.  My comment was to suggest that I felt like keeping this test was worthwhile, as I care about this use case, and the ref_chain logic is of purely instrumental value.  I suggested having a separate test chain logic as a compromise, but if you're fine with just this test then I'm happy to just keep thing as they are.